code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/ServiceDiscoveryLoadBalancerTargetAddPostListener.java

```diff
@@ -89,70 +92,88 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
         return null;
     }
 
-    private List<Pair<ServiceExposeMap, Long>> getInstanceToLBServiceMap(ProcessState state, ProcessInstance process) {
-        List<Pair<ServiceExposeMap, Long>> exposeMapToLBService = new ArrayList<>();
+    private List<Pair<ServiceExposeMap, Service>> getInstanceToLBServiceMap(ProcessState state, ProcessInstance process) {
+        List<Pair<ServiceExposeMap, Service>> exposeMapToLBService = new ArrayList<>();
         if (process.getName().equalsIgnoreCase(InstanceConstants.PROCESS_START)) {
-            Instance instance = (Instance) state.getResource();
-            List<ServiceExposeMap> maps = objectManager.mappedChildren(
-                    objectManager.loadResource(Instance.class, instance.getId()),
-                    ServiceExposeMap.class);
-            if (maps.isEmpty()) {
-                // handle only instances that are the part of the service
-                return exposeMapToLBService;
+            getLBServiceForInstance(state, exposeMapToLBService);
```

Comment:
I think it might be more clear if we pass the casted state.getResource() into these functions so the method signatures for these make more sense.